components/profile/activityTab/Offers.vue

```diff
@@ -0,0 +1,171 @@
+              class="text-k-grey"
+            >
+              ({{ counts[filter.id] || '' }})
+            </span>
+          </div>
+        </NeoButton>
+      </div>
+    </div>
+
+    <hr class="mb-10 mt-0">
+
+    <div>
+      <ResponsiveTable
+        :no-results-main="$t('activity.noResults')"
+        :no-results-sub="$t('activity.noResultsSub')"
+        :items="offers"
+        :show-no-results="!offers.length || !activeFilters.length"
+        :loading="loading"
+      >
+        <template #columns>
+          <div class="flex-1">
+            <span>{{ $t('activity.event.item') }}</span>
+          </div>
+          <div class="w-1/12">
+            <span>{{ $t('activity.event.event') }}</span>
+          </div>
+          <div class="flex-1">
+            <span>{{ $t('activity.event.amount') }}</span>
+          </div>
+          <div class="flex-1">
+            <span>{{ $t('activity.event.to') }}</span>
+          </div>
+          <div class="flex-1">
+            <span>{{ $t('activity.event.time') }}</span>
+          </div>
+          <div class="flex-1">
+            <span />
+          </div>
+        </template>
+
+        <template #rows="{ variant }">
+          <ProfileActivityTabOfferRow
+            v-for="item in offers"
+            :key="item.id"
+            data-testid="offer-item-row"
+            :offer="item"
+            :variant="variant as unknown"
+            @select="() => {
+              selectedOffer = item
+              isOfferModalOpen = true
+            }"
+          />
+        </template>
+      </ResponsiveTable>
+    </div>
+  </div>
+
+  <OfferOverviewModal
+    v-model="isOfferModalOpen"
+    :offer="selectedOffer"
+    @close="() => {
+      selectedOffer = undefined
+      isOfferModalOpen = false
+    }"
+  />
+</template>
+
+<script lang="ts" setup>
+import { NeoButton } from '@kodadot1/brick'
+
+const props = defineProps<{
+  id: string
+}>()
+
+const route = useRoute()
+const { replaceUrl } = useReplaceUrl()
+
+const selectedOffer = ref<NFTOfferItem>()
+const isOfferModalOpen = ref(false)
+const counts = ref<{ incoming: number, outgoing: number }>()
+
+const toBoolean = (param: unknown): boolean => param === 'true'
+
+const filters = ref([{
+  id: 'outgoing',
+  icon: 'arrow-up',
+  active: toBoolean(route.query.outgoing),
+}, {
+  id: 'incoming',
+  icon: 'arrow-down',
+  active: toBoolean(route.query.incoming),
+}])
+
+const syncQuery = computed(() => Object.fromEntries(filters.value.map(filter => [filter.id, filter.active])))
+const activeFilters = computed(() => Object.keys(syncQuery.value).filter(queryKey => syncQuery.value[queryKey]))
+
+if (!activeFilters.value.length) {
+  filters.value = filters.value.map(filter => filter.id === 'outgoing' ? ({ ...filter, active: true }) : filter)
+}
+
+const where = computed(() => {
+  const conditions = [] as Record<string, unknown>[]
+
+  const outgoing = {
+    caller_eq: props.id,
+    status_eq: 'ACTIVE',
+  }
```

Comment:
I would display all the offers including active and expired ones, so I can withdraw the deposit from the expired offer.

I guess the deposit in the offer will not automatically return when it becomes expires.

Reply:
done